Sources/PackageModel/UserToolchain.swift

```diff
@@ -328,7 +328,11 @@ public final class UserToolchain: Toolchain {
         swiftSDK: SwiftSDK,
         environment: EnvironmentVariables
     ) throws -> [String] {
-        let swiftCompilerFlags = swiftSDK.toolset.knownTools[.swiftCompiler]?.extraCLIOptions ?? []
+        var swiftCompilerFlags = swiftSDK.toolset.knownTools[.swiftCompiler]?.extraCLIOptions ?? []
+
+        if let linker = swiftSDK.toolset.knownTools[.linker]?.path {
+            swiftCompilerFlags += ["-Xclang-linker", "--ld-path=\(linker)"]
```

Comment:
I'm unsure about passing this only via `-Xclang-linker`. This means that only Clang driver will see that we switched to a different linker, but the Swift Driver won't. In the latter there's a certain amount of logic around this, for example see https://github.com/apple/swift-driver/blob/e98ddbdab89730d039a461ca134f0e5ec20e55dc/Sources/SwiftDriver/Jobs/WindowsToolchain%2BLinkerSupport.swift#L45 and https://github.com/apple/swift-driver/blob/e98ddbdab89730d039a461ca134f0e5ec20e55dc/Sources/SwiftDriver/Jobs/GenericUnixToolchain%2BLinkerSupport.swift#L75. If you're doing this because the Swift Driver doesn't support `--ld-path`, I'd rather see that option added there.

Reply:
hmm IIUC the Swift driver only cares about the linker flavor, not the path — we're not attempting to determine the flavor at all anymore. The `-Xclang-linker --ld-path=` strategy would indeed break if the Swift driver started invoking the linker directly, but from what I can tell that doesn't happen at the moment.

Reply:
For what it's worth I do agree that the Swift Driver should handle linker customization better: in addition to not having an `--ld-path` option, atm it doesn't even seem to respect `tools-directory` for the linker (despite `swiftc -h` claiming that it does). I'm considering making a PR to fix both of these things, after which we can avoid the `-Xclang-linker` here, though I think keeping it in for now is okay since the Swift driver doesn't look at the linker path at all yet.

Reply:
cc @artemcm

Reply:
Created https://github.com/apple/swift/pull/67956 and https://github.com/apple/swift-driver/pull/1416 to add `--ld-path` to the driver

Reply:
Both of those are merged now, is this PR now the only thing remaining for this fix to fully land?

Reply:
@MaxDesiatov I believe so!

Reply:
Great, waiting for CI to pass here then.